File: src/uimd/dialogs/message_box_ok.py
"""Standard one-button message box."""

try:
    from uimd.dialogs.message_box import MessageBoxBehavior
    from uimd.dialogs.message_box_ok_ui import MessageBoxOkUI
except ImportError:
    from message_box import MessageBoxBehavior
    from message_box_ok_ui import MessageBoxOkUI


class MessageBoxOk(MessageBoxBehavior, MessageBoxOkUI):
    def __init__(self, header="Message", message="", on_close=None):
        super().__init__()
        self._configure_message_box(
            header=header,
            message=message,
            on_close=on_close,
            result_by_button={"ok_btn": "ok"},
            default_result="ok",
        )

File: java/src/main/java/uimd/MessageBoxOk.java
package uimd;

import java.util.Map;
import java.util.function.Consumer;

public final class MessageBoxOk extends MessageBoxOkUI
{
    private final MessageBoxBehavior behavior;

    public MessageBoxOk()
    {
        this("Message", "", null);
    }

    public MessageBoxOk(String header, String message)
    {
        this(header, message, null);
    }

    public MessageBoxOk(String header, String message, Consumer<String> onClose)
    {
        behavior = new MessageBoxBehavior(
            dialog_header,
            this.message,
            header,
            message,
            onClose,
            Map.of("ok_btn", "ok"),
            "ok");
    }

    public String result()
    {
        return behavior.result();
    }

    public boolean closed()
    {
        return behavior.closed();
    }

    public int run(String[] arguments)
    {
        return GeneratedWindowRuntime.runGeneratedWindow(this, runtimeOptions(), arguments);
    }

    @Override
    public GeneratedWindowRuntimeOptions runtimeOptions()
    {
        return behavior.configure(super.runtimeOptions());
    }

    @Override
    protected void onOkBtnClick()
    {
        behavior.handleButton("ok_btn");
    }
}
